pkg/resolution/resolver/framework/interface.go

```diff
@@ -43,13 +43,25 @@ type Resolver interface {
 	// request and should return an error if any are missing or invalid.
 	ValidateParams(ctx context.Context, params []pipelinev1.Param) error
 
+	// ValidateUrl is given the url from a resource
+	// request and should return an error if the resolver cannot resolve it.
+	ValidateUrl(ctx context.Context, url string) error
+
 	// Resolve receives the parameters passed via a resource request
 	// and returns the resolved data along with any annotations
 	// to include in the response. If resolution fails then an error
 	// should be returned instead. If a resolution.Error
 	// is returned then its Reason and Message are used as part of the
 	// response to the request.
 	Resolve(ctx context.Context, params []pipelinev1.Param) (ResolvedResource, error)
+
+	// ResolveUrl receives the url passed via a resource request
+	// and returns the resolved data along with any annotations
+	// to include in the response. If resolution fails then an error
+	// should be returned instead. If a resolution.Error
+	// is returned then its Reason and Message are used as part of the
+	// response to the request.
+	ResolveUrl(ctx context.Context, url string) (ResolvedResource, error)
```

Comment:
IIRC, we originally wanted params to take precedence over url params if both were used. Is this still the case?